### PR TITLE
[FLINK-19947][Connectors / Common]Support sink parallelism configuration for Print connector

### PR DESCRIPTION
**Pull request by zhuxiaoshang**
## What is the purpose of the change

Support sink parallelism configuration for Print connector

## Brief change log
 - Support sink parallelism configuration for Print connector

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (docs)

### CODE REVIEW

**Comment by zhuxiaoshang** [comment]
cc @JingsongLi @wuchong

**Comment by wuchong** [comment]
cc @fsk119

**Review by fsk119 — CHANGES_REQUESTED** [review]
Thanks for your contribution. It looks good to me except no test in this pr.
You can add test in `PrintConnectorITCase` to verify the correctness. The `PrintSinkOutputWriter` will also print the subtask id, e.g. `test_print:1>` when parallelism is > 1.

File: flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/runtime/stream/table/PrintConnectorITCase.java [hunk 1]
```diff
@@ -77,6 +77,31 @@ public void testStandardError() throws Exception {
...
+				"f0 int," +
+				"f1 double" +
+				") with (" +
+				"'connector' = 'print'," +
+				"'print-identifier' = 'test_print'," +
+				"'sink.parallelism' = '2'," +
+				"'standard-error'='false')");
+		DataType type = tEnv().from("print_t").getSchema().toRowDataType();
+		Row row = Row.of(1, 1.1);
+		tEnv().fromValues(type, Arrays.asList(row)).executeInsert("print_t").await();
```
Comment (fsk119) [inline flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/runtime/stream/table/PrintConnectorITCase.java:92]: `Arrays.asList` -> `Collections.singletonList`

**Review by fsk119 — COMMENTED** [review]
LGTM except the minor problem.

**Review by fsk119 — APPROVED** [review]
LGTM

File: flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/PrintTableSinkFactory.java [hunk 2]
```diff
@@ -95,19 +96,22 @@ public DynamicTableSink createDynamicTableSink(Context context) {
...
+				options.get(STANDARD_ERROR),
+				options.getOptional(FactoryUtil.SINK_PARALLELISM).orElse(null));
 	}
 
 	private static class PrintSink implements DynamicTableSink {
 
 		private final DataType type;
 		private final String printIdentifier;
 		private final boolean stdErr;
+		private final Integer parallelism;
```
Comment (wuchong) [inline flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/factories/PrintTableSinkFactory.java:108]:
```suggestion
		private final @Nullable Integer parallelism;
```

Would be better to add `@Nullable` annotation. 

See the Flink Code Style Guideline: https://flink.apache.org/contributing/code-style-and-quality-java.html#java-optional

**Review by wuchong — APPROVED** [review]
LGTM.

Will merge this once Azure is green.

**Comment by wuchong** [comment]
The build compile is failed, please have a look @zhuxiaoshang

**Comment by zhuxiaoshang** [comment]
@flinkbot run azure

**Comment by zhuxiaoshang** [comment]
@wuchong Azure is green

**Pull request merged**